source/solvers/heat_transfer.cc

```diff
@@ -486,20 +488,32 @@ HeatTransfer<dim>::assemble_local_system_matrix(
           simulation_control->get_current_time() >
             this->simulation_parameters.post_processing.initial_time)
         {
-          scratch_data.reinit_velocity(velocity_cell,
+          scratch_data.reinit_velocity(fd_cell,
                                        *multiphysics->get_time_average_solution(
                                          PhysicsID::fluid_dynamics),
                                        this->simulation_parameters.ale);
         }
       else
         {
-          scratch_data.reinit_velocity(velocity_cell,
+          scratch_data.reinit_velocity(fd_cell,
                                        *multiphysics->get_solution(
                                          PhysicsID::fluid_dynamics),
                                        this->simulation_parameters.ale);
         }
     }
 
+  if (!this->simulation_parameters.physical_properties_manager
+         .density_is_constant())
+    {
+      if (!multiphysics->fluid_dynamics_is_block())
+        scratch_data.reinit_pressure(
+          fd_cell, *multiphysics->get_solution(PhysicsID::fluid_dynamics));
+      else
+        scratch_data.reinit_pressure(fd_cell,
+                                     *multiphysics->get_block_solution(
```

Comment:
It's better to  move all the reinits to a single reinit which would be reinit_fluid_dynamics,. Otherwise the call to fe_values reinit will get prohibitively expensive..

Reply:
You're right, I'll do so tomorrow!
I'll also check if it is more costly to add an `if` inside to get the pressure values or if the `get_function_values()` is not to bad. If it is not too bad (which I don't think is the case), I'll leave it without the `if` is that okay ?

Reply:
Yup sounds good